aiohttp_client_cache/backends/dynamodb.py

```diff
@@ -1,148 +1,179 @@
-        connection: ServiceResource = None,
-        region_name: str = 'us-east-1',
-        read_capacity_units: int = 1,
-        write_capacity_units: int = 1,
-        **kwargs,
+        namespace: str,
+        key_attr_name: str,
+        val_attr_name: str,
+        create_if_not_exists: bool,
+        context: ResourceCreatorContext,
     ):
+        self.table_name = table_name
         self.namespace = namespace
-        self.connection = connection or boto3.resource(
-            'dynamodb', region_name=region_name, **kwargs
-        )
-
-        # Create the table if it doesn't already exist
-        try:
-            self.connection.create_table(
-                AttributeDefinitions=[
-                    {
-                        'AttributeName': 'namespace',
-                        'AttributeType': 'S',
-                    },
-                    {
-                        'AttributeName': 'key',
-                        'AttributeType': 'S',
-                    },
-                ],
-                TableName=table_name,
-                KeySchema=[
-                    {'AttributeName': 'namespace', 'KeyType': 'HASH'},
-                    {'AttributeName': 'key', 'KeyType': 'RANGE'},
-                ],
-                ProvisionedThroughput={
-                    'ReadCapacityUnits': read_capacity_units,
-                    'WriteCapacityUnits': write_capacity_units,
-                },
-            )
-        except ClientError:
-            pass
-
-        self._table = self.connection.Table(table_name)
-        self._table.wait_until_exists()
-
-    def _scan_table(self) -> Dict:
-        return self._table.query(
-            ExpressionAttributeValues={':Namespace': self.namespace},
-            ExpressionAttributeNames={'#N': 'namespace'},
-            KeyConditionExpression='#N = :Namespace',
-        )
-
-    @staticmethod
-    def unpickle(response_item: Dict) -> ResponseOrKey:
-        return BaseCache.unpickle((response_item or {}).get('value'))
+        self.key_attr_name = key_attr_name
+        self.val_attr_name = val_attr_name
+        self.create_if_not_exists = create_if_not_exists
+        self.context = context
+        self._table = None
+
+    async def get_table(self):
+        if not self._table:
+            # Re-use the service resource if it's already been created
+            if self.context.cls:
+                conn = self.context.cls
+            # otherwise create
+            else:
+                # should we try to call aexit later if we auto enter here?
+                conn = await self.context.__aenter__()
+
+            self._table = await conn.Table(self.table_name)
+            if self.create_if_not_exists:
+                try:
+                    await conn.create_table(
+                        AttributeDefinitions=[
+                            {
+                                'AttributeName': self.key_attr_name,
+                                'AttributeType': 'S',
```

Comment:
I simplified the table to just include a single key. Originally, I tried adding TTL too, but that complicated things so much, that I opted to try to KISS for an initial async impl. My interpretation of dynamo's docs are that having the namespace in its own hash key means that all items for that namespace will effectively go to the same partition, which seems bad. I would think the items should be distributed across partitions, and so opted for this flatter impl where the namespace is just a prefix added to the key.

Reply:
I wasn't aware of that behavior with namespaces and partitioning. That sounds good to me.